src/Dropdown.php

```diff
@@ -4,359 +4,657 @@
+     * Whether to automatically close the dropdown when the user clicks inside of the dropdown.
+     *
+     * @param bool $value If set to `true`, the dropdown will be closed when the user clicks inside of the dropdown. If
+     * set to `false`, no attribute will be set.
+     *
+     * @return self A new instance with the specified auto-close setting.
+     */
+    public function autoCloseOnClickInside(bool $value = true): self
+    {
+        return $this->addToggleAttributes(['data-bs-auto-close' => $value === true ? 'inside' : null]);
+    }
 
-        return Html::openTag($this->tagName, $options);
+    /**
+     * Whether to automatically close the dropdown when the user clicks outside of the dropdown or the dropdown toggle
+     * button.
+     *
+     * @param bool $value If set to `true`, the dropdown will be closed when the user clicks outside of the dropdown. If
+     * set to `false`, no attribute will be set.
+     *
+     * @return self A new instance with the specified auto-close setting.
+     */
+    public function autoCloseOnClickOutside(bool $value = true): self
+    {
+        return $this->addToggleAttributes(['data-bs-auto-close' => $value === true ? 'outside' : null]);
     }
 
     /**
-     * @throws InvalidConfigException
-     * @throws JsonException
-     * @return string
+     * Replaces all existing CSS classes of the dropdown component with the provided ones.
+     *
+     * Multiple classes can be added by passing them as separate arguments. `null` values are filtered out
+     * automatically.
+     *
+     * @param BackedEnum|string|null ...$value One or more CSS class names to set. Pass `null` to skip setting a class.
+     * For example:
+     *
+     * ```php
+     * $dropdown->class('custom-class', null, 'another-class', BackGroundColor::PRIMARY());
+     * ```
+     *
+     * @return self A new instance with the specified CSS classes set.
      */
-    public function render(): string
+    public function class(BackedEnum|string|null ...$value): self
     {
-        if ($this->tagName) {
-            return Html::closeTag($this->tagName);
-        }
+        $new = clone $this;
+        $new->cssClasses = $value;
 
-        $layout = $this->prepareDropdownLayout();
+        return $new;
+    }
 
-        if (is_array($this->items)) {
-            /** @var Ul $layout */
-            return $this->renderItems($layout, $this->items);
-        }
-        /** @var CustomTag $layout */
-        return $layout
-            ->content((string) $this->items)
-            ->render();
+    /**
+     * Whether to render the dropdown in a container `<div>` tag.
+     *
+     * @param bool $value Whether to render the dropdown in a container `<div>` tag. By default, it will be rendered in
+     * a container `<div>` tag. If set to `false`, the container will not be rendered.
+     *
+     * @return self A new instance with the specified container setting.
+     */
+    public function container(bool $value): self
```

Comment:
What case when dropdown should be generated without container?

Reply:
https://getbootstrap.com/docs/5.3/components/navbar/#supported-content When used in a NavBar.